java/spring/security/injection/tainted-file-path.yaml

```diff
@@ -0,0 +1,47 @@
+rules:
+  - id: tainted-file-path
+    languages:
+      - java
+    severity: ERROR
+    message: Detected user input controlling a file path. An attacker could control
+      the location of this file, to include going backwards in the directory
+      with '../'. To address this, ensure that user-controlled variables in file
+      paths are sanitized. You may also consider using a utility method such as
+      org.apache.commons.io.FilenameUtils.getName(...) to only retrieve the file
+      name from the path.
+    metadata:
+      cwe: "CWE-23: Relative Path Traversal"
+      owasp:
+        - A01:2021 - Broken Access Control
+        - A05:2017 - Broken Access Control
+      references:
+        - https://owasp.org/www-community/attacks/Path_Traversal
+      category: security
+      technology:
+        - java
+        - spring
+    mode: taint
+    pattern-sources:
+      - patterns:
+          - pattern-either:
+              - pattern-inside: |
+                  $METHODNAME(..., @$REQ($LOOKUP) String
+                  $SOURCE,...) {
+                    ...
+                  }
+              - pattern-inside: |
+                  $METHODNAME(..., @$REQ String
+                  $SOURCE,...) {
+                    ...
+                  }
+          - metavariable-regex:
+              metavariable: $REQ
+              regex: (RequestBody|PathVariable|RequestParam)
+          - pattern: $SOURCE
+    pattern-sinks:
+      - pattern-either:
+          - pattern-either:
+              - pattern: new FileInputStream(...)
```

Comment:
we might want to make these explicit, e.g. `java.io.FileInputStream`

Reply:
I'm not actually clear on the behavior differences between the two. Should I just add both? Or only the explicit ones?